### PR TITLE
Fix errors found by yetus (markdownlint plugin)

### PR DESCRIPTION
**Pull request by rene**
Fix errors found by yetus (markdownlint plugin)

Fix all errors found by the plugin "markdownlint" from yetus.

Signed-off-by: Renê de Souza Pinto <rene@renesp.com.br>

PS1: Separating these changes from PR https://github.com/lf-edge/eve/pull/2831
PS2: No source files are changed by this PR.

.github/CODE_OF_CONDUCT.md
CONTRIBUTING.md
docs/BASEIMAGE-UPDATE.md
docs/BUILD.md
docs/CI-CD.md
docs/CLEANUP.md
docs/CUSTOM-BUILD.md
docs/DEBUGGING.md
docs/ECO-METADATA.md
docs/ECOS.md
docs/EVE-IMAGE-REPRODUCIBILITY.md
docs/EVE-IMAGE-SOURCES.md
docs/FAQ.md
docs/HYPERVISORS.md
docs/LOGGING.md
docs/NETWORK-MODELS.md
docs/SECURITY.md
pkg/pillar/docs/domainmgr.md
pkg/pillar/docs/volumemgr.md
pkg/xen/README.md

### CODE REVIEW

File: docs/DEBUGGING.md [hunk 1]
```diff
@@ -83,192 +83,192 @@ If BootReason indicates the system crashed due to a touch file, this means that
 
 The output will be a single long line with carriage returns and tabs as escaped characters. Fix it by replacing the characters and dumping them to a file:
 
-```sh
+```text
```
Comment (eriknordmark) [inline docs/DEBUGGING.md:86]: I guess this didn't get rid of the complaints? 
It isn't important but a bit annoying that there isn't a way to get markdownlint to accept verbatim input.
But I'll let it slide.

**Review by eriknordmark — COMMENTED** [review]
I don't understand why docs/EVE-IMAGE-SOURCES.md and docs/EVE-IMAGE-REPRODUCIBILITY.md need fixing because since @deitch just added those hence they ran through the full yetus checks. So why should blanks complain? Does the blanks run by yetus run with different options than when you run it manually?

File: docs/DEBUGGING.md (hunk 1 above)
Comment (rene) [reply docs/DEBUGGING.md:86]: Yes @eriknordmark, I changed to "text" as you suggested in the original PR. Regarding hard tabs, from markdownlint documentation for the Rule MD010 (no hard tabs - https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md):

`You have the option to exclude this rule for code blocks and spans. To do so, set the code_blocks parameter to false. Code blocks and spans are included by default since handling of tabs by Markdown tools can be inconsistent (e.g., using 4 vs. 8 spaces).`

Well, I would still insist to keep the spaces in our documentation files in order to avoid an eventual mishandling of these tabs for any markdown processing tool. Also, for code blocks of other languages we must want to have this checking in order to avoid mixes of hard tabs vs spaces (this is quite annoying and usually common to happen).

In summary, if you agree, let's keep this PR as it is...

**Comment by rene** [comment]
> I don't understand why docs/EVE-IMAGE-SOURCES.md and docs/EVE-IMAGE-REPRODUCIBILITY.md need fixing because since @deitch just added those hence they ran through the full yetus checks. So why should blanks complain? Does the markdownlink run by yetus run with different options than when you run it manually?

https://github.com/lf-edge/eve/pull/2810/commits/c7d451776dc8c349720b95f57b27662bef42a67a

It looks like it was merged apart from the errors, or am I wrong?

**Comment by eriknordmark** [comment]
> It looks like it was merged apart from the errors, or am I wrong?

There were <10 yetus complaints about fenced etc which @deitch argued that markdownlint is incorrect, but you are fixing lots more lines than those. That is the part I don't understand.

**Comment by rene** [comment]
> > It looks like it was merged apart from the errors, or am I wrong?
> 
> There were <10 yetus complaints about fenced etc which @deitch argued that markdownlint is incorrect, but you are fixing lots more lines than those. That is the part I don't understand.

For me, these were the errors pointed by markdownlint to these files:
```
docs/EVE-IMAGE-REPRODUCIBILITY.md:83:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-REPRODUCIBILITY.md:135:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-REPRODUCIBILITY.md:146:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-REPRODUCIBILITY.md:174:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-REPRODUCIBILITY.md:251:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-REPRODUCIBILITY.md:265:0:MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
docs/EVE-IMAGE-REPRODUCIBILITY.md:265:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-REPRODUCIBILITY.md:271:0:MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
docs/EVE-IMAGE-REPRODUCIBILITY.md:271:0:MD046/code-block-style Code block style [Expected: indented; Actual: fenced]
docs/EVE-IMAGE-SOURCES.md:37:100:MD034/no-bare-urls Bare URL used [Context: "https://github.com/lf-edge/eve..."]
docs/EVE-IMAGE-SOURCES.md:41:0:MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "* For Docker images, run the `..."]
docs/EVE-IMAGE-SOURCES.md:42:0:MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:42:0:MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:45:0:MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:46:0:MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "* For release artifacts, mount..."]
docs/EVE-IMAGE-SOURCES.md:47:0:MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:47:0:MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:52:79:MD034/no-bare-urls Bare URL used [Context: "https://github.com/lf-edge/eve"]
docs/EVE-IMAGE-SOURCES.md:66:111:MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
docs/EVE-IMAGE-SOURCES.md:70:0:MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:157:0:MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
docs/EVE-IMAGE-SOURCES.md:158:0:MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 3]
docs/EVE-IMAGE-SOURCES.md:253:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:254:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:255:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:273:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:297:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:399:0:MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
docs/EVE-IMAGE-SOURCES.md:412:0:MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
docs/EVE-IMAGE-SOURCES.md:653:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:654:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:655:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:656:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:657:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:658:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:660:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:661:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:662:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:665:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:666:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:667:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:668:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:669:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:670:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:671:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:672:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:673:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:674:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:675:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:676:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:678:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:679:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:680:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:682:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:686:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:688:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:689:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:690:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:691:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:693:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:694:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:695:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:697:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:698:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:699:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:700:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:704:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:705:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:707:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:708:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:709:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:711:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:712:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:716:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:718:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:719:1:MD010/no-hard-tabs Hard tabs [Column: 1]
docs/EVE-IMAGE-SOURCES.md:741:87:MD047/single-trailing-newline Files should end with a single newline character
```

**Comment by eriknordmark** [comment]
> For me, these were the errors pointed by markdownlint to these files:

Which raises the question about what differs in the config/options to markdownlint when you run in manually and when the build runs it via yetus, because there were e.g., no complaints about 'no hard tabs' in the run on the PR which introduced these files.
So your manual run is more picky than the build, hence we can't use that as the reference.

**Comment by rene** [comment]
> > For me, these were the errors pointed by markdownlint to these files:
> 
> Which raises the question about what differs in the config/options to markdownlint when you run in manually and when the build runs it via yetus, because there were e.g., no complaints about 'no hard tabs' in the run on the PR which introduced these files. So your manual run is more picky than the build, hence we can't use that as the reference.

I'm running yetus locally with make (make yetus), which runs the following command:

```sh
docker run -it --rm -v $(CURDIR):/src:delegated -v /tmp:/tmp apache/yetus:0.14.0 \
345         --basedir=/src \
346         --dirty-workspace \
347         --empty-patch \
348         --plugins=all
```

On Github actions, yetus runs through [yetus-test-patch-action](https://github.com/apache/yetus-test-patch-action) with the following configuration:  https://github.com/lf-edge/eve/blob/master/.github/workflows/yetus.yml#L28

From the [documentation](https://yetus.apache.org/documentation/0.14.0/precommit/robots/githubactions/) the command line equivalence for this action is:

```
          --basedir=/src
          --buf-basedir=/src/api/proto
          --build-tool=nobuild
          --continuous-improvement=true
          --github-token=${{ secrets.GITHUB_TOKEN }}
          --patch-dir=./out
          --revive-config=.revive.toml
```

Checking arguments that appears only at "one side":

- _dirty-workspace_: Allow the local git workspace to have uncommitted changes (ok)
- _empty-patch_: Create a summary of the current source tree (ok)
- _continuous improvement_: will not fail with errors already present in the repository, but only for errors added by PRs ( that's needed otherwise yetus would fail for all PRs because of current errors)
- _buf-basedir_: starting directory to run protobuf linter (adding to Makefile didn't changed the results)
- _build-tool_:  Pick which build tool to focus around (default: autodetect from 'nobuild')
- _revive-config_: Configuration file for revive (.revive.toml is already the default name)
- _github-token_: Github token (ok)
-  _patch-dir_: The directory for working and output files (default '/tmp/test-patch-unknown/pid') (ok)

For me these differences would not impact on markdownlint output, but I might be missing something...

**Comment by eriknordmark** [comment]
> For me these differences would not impact on markdownlint output, but I might be missing something...

Can you get the output from yetus for https://github.com/lf-edge/eve/pull/2810 from https://github.com/lf-edge/eve/actions/runs/3095895201 and see if there is any indication why it didn't flag the whitespace? The scan output is in https://github.com/lf-edge/eve/suites/8392785210/artifacts/370770663

**Comment by rene** [comment]
> 

@eriknordmark , just checked the outputs, but first let me clarify (to avoid any confusion), that you are saying "blanks errors", but these errors regarding "no-hard-tabs" they are issued by markdownlint plugin, instead of blanks plugin. With that said, I checked the outputs from the link you asked: https://github.com/lf-edge/eve/actions/runs/3095895201

Indeed, we cannot see the no-hard-tabs errors in the annotations (which points only 10 errors). However, if we click on yetus to see the detailed output, all the errors are there, without pointing to the file tough:

https://github.com/lf-edge/eve/actions/runs/3095895201/jobs/5010814255#step:4:244

It seems a (miss?)behavior from GitHub than an error on our side. I checked the yetus command line from my PR against Avi's PR... they are exactly the same... but again, although these errors are not shown on Annotations, they are in fact pointed out by yetus... I couldn't check more deeper because I don't have access to log files generated by yetus on GitHub (those written to /github/workspace/out/, for instance, /github/workspace/out/results-markdownlint.txt)

**Comment by rene** [comment]
Updates in this PR:
- Rebased with master

**Comment by eriknordmark** [comment]
@rene I chatted with @rvs and he said you can reach out to him on how yetus handles its output to create the annotations. But it sounds like independently of that we should proceed with this PR.

**Review by eriknordmark — APPROVED** [review]
LGTM
